src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -317,13 +318,21 @@ public function __construct(A $a)
 {
 }
 
-class F implements EInterface
+interface IInterface
+{
+}
+
+class I implements IInterface
+{
+}
+
+class F extends I implements EInterface
```

Comment:
you also needs the case of an interface extending another interface (and the class implementing the child one), to be sure it works fine too